### PR TITLE
Add visual studio coverage test for MultipleProjects

### LINKED ISSUE #3506: Add more complex integration tests for the C# plugin
in #3509 :
- [x] Create a solution with 2 C# projects, each with its own test solution
- [x] Add integration test for importing measures and highlighting
- [x] Add integration test for importing issues

in #3512  :
- [x] Add integration test for importing OpenCover coverage

left to do:
- [x] Add integration test for importing VS Coverage

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Fixes #3506

### CODE REVIEW

File: its/src/test/java/com/sonar/it/csharp/CoverageTest.java [hunk 1]
```diff
@@ -137,6 +128,21 @@ public void visual_studio() throws Exception {
...
   }
 
+  @Test
+  public void visual_studio_on_MultipleProjects() throws Exception {
+    BuildResult buildResult = analyzeMultipleProjectsTestProject("sonar.cs.vscoveragexml.reportsPaths", "VisualStudio.coveragexml");
+
+    assertThat(buildResult.getLogs()).contains(
+      "Sensor C# Tests Coverage Report Import",
+      "Coverage Report Statistics: 4 files, 3 main files, 3 main files with coverage, 1 test files, 0 project excluded files, 0 other language files.");
+
```
Comment (costin-zaharia-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/CoverageTest.java:138]: It's interesting to see the difference with opencover:

On opencover:

> Coverage Report Statistics: 5 files, 3 main files, 3 main files with coverage, 2 test files, 0 project excluded files, 0 other language files

On VS:

> Coverage Report Statistics: 4 files, 3 main files, 3 main files with coverage, 1 test files, 0 project excluded files, 0 other language files.

I would expect to see 2 test files also for VS (`FirstProjectTests.cs` and `FirstClassTests.cs`)

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
Overall looks good but it would be interesting to know why Visual Studio reports 1 test file instead of 2.

File: its/src/test/java/com/sonar/it/csharp/CoverageTest.java (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/CoverageTest.java]: good catch; I noticed this but I only ignored it saying that "meh, vs coverage is weird / stuped" but *I was wrong*.

The problem was that for `SecondProjectTests\FirstClassTests.cs` I missed making the path relative (the manual step)

File: its/src/test/java/com/sonar/it/csharp/CoverageTest.java [hunk 2]
```diff
@@ -196,7 +196,7 @@ private BuildResult analyzeProject(String projectName, String... keyValues) thro
 
     orchestrator.executeBuild(beginStep);
 
-    TestUtils.runMSBuild(orchestrator, projectDir, "/t:Rebuild");
+    TestUtils.runMSBuild(orchestrator, projectDir, "/t:Restore,Rebuild");
```
Comment (andrei-epure-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/CoverageTest.java:199]: this was needed after rebasing

File: its/src/test/java/com/sonar/it/csharp/CoverageTest.java (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/CoverageTest.java]: Looks good now, the match is almost perfect 😄 

```
OpenCover:
Coverage Report Statistics: 5 files, 3 main files, 3 main files with coverage, 2 test files, 0 project excluded files, 0 other language files
VS:
Coverage Report Statistics: 5 files, 3 main files, 3 main files with coverage, 2 test files, 0 project excluded files, 0 other language files.
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**